Map/Source/SquareMap.cpp

```diff
@@ -22,3 +22,54 @@ SquareMap::SquareMap(const int mapSize)
         _fieldList.emplace_back(std::move(column));
     }
 }
+
+bool SquareMap::isMovePossible(const char pressedKey, int &x, int &y)
+{
+    switch(pressedKey)
+    {
+        case 'w':
+        {
+            if(y + 1 < _fieldList.size() && _fieldList.at(x).at(y + 1) -> getType() != FieldType::Wall)
+            {
+                //zmienić typ pola, na którym stoi gracz
```

Comment:
komentarze zbedne

Reply:
funkcja zwraca bool, a nie widze tutaj zadnego return true. albo return false :)

Reply:
Domyślam się że nie skompilowałeś kodu przed commite, bo przez nasze flagi kompilacji brak returna w takiej funkcji spowodował by błąd i kod by się nie skompilował :)

Reply:
warto każdy z tych casów wrzucić do funkcji, no isMoveForwardPosiible itd.

Reply:
@marcinwozniak0 Można zrobić z tej metody isFieldAvailable i przyjmować po prostu x i y

Reply:
Czyli mam zrobić 4 funkcje bool isMoveRightPossible itp. które przyjmą współrzędne x i y pola na którym aktualnie stoi gracz i powiedzą czy można wykonać ruch w tym kierunku?

Reply:
Ja się tak wtrące jako osoba z drugiego projektu, która z ciekawości do Was zajrzała.

1. argumenty funkcji X, Y --> nie mówią mi to nic, zakładam, że są to współrzędne, ale nie wiem czy tam gdzie jest gracz, czy tam gdzie jest pole, na które chce iść czy co
2. Nie macie odzielonego GUI od logiki, duży błąd architektury. Powinniście wystawić move(), która pozwala zrobić ruch, w środku mieć isMovePossible(), który sprawdza czy można zrobić ruch. Jedno i drugie powinno przyjmować enum direction {left, right, ...}. WASD powinno być nieznane dla logiki. Logika powinna operować na kierunkach abstrakcyjnych. GUI powinno przekonwertować WASD na kierunki. A co jeśli będziemy chcieli zmienić sterowanie na strzałki? Nagle taka zmiana powoduje zmianę w logice sprawdzania pól, a przecież to są niezależne rzeczy od siebie.
3. && --> and
4. Macie buga w tym kodzie. Rzuci Wam wyjątkiem, bo sprawdzacie w tym samym momencie at(y+1) co y + 1 < _fieldList.size(). Powinniście zrobić to w dwóch ifach następujących po sobie, pierw if size, potem if wall.
5. Ogólnie polecamstworzyć sobie strukturę, która jest współrzędnymi i przechowuje X oraz Y. Bo nie macie momentu, w któ©ym występują one odzielnie. Dzięki temu każdy wie o co chodzi:

int& x, int& y --> nic nam nie mówi
coordinates player --> współrzędne gracza, czyli zapewne gdzie teraz stoi

Tak samo stosować x i y czy row and column?

Reply:
@mnurzyns 
2. Zagadza się, tak to będzie wyglądać. Ale zadanie polegało na stworzeniu na razie tylko tej metody. 
4. Nie zgadzam się, bo jeśli  y/x + 1 będzie większe bądz równe _fieldList.size() to dalsza część ifa się nie wykona i nie dojdzie do rzucenia wyjątku

@dchoptiany zastosuj resztę uwag Marcina. Olej jedynie robienie enumów, zrobimy to w osobnym zadaniu. Jak poprawisz to daj mi znać na slacku

Reply:
@marcinwozniak0 
4. W większości przypadków tak, zdefiniowany and jest left-handed. Aczkolwiek dobrą praktyką (w C++, nie w C) jest rozbijanie tego na dwa ify.